framework/include/problems/FEProblemBase.h

```diff
@@ -431,8 +432,18 @@ class FEProblemBase : public SubProblem, public Restartable
    * local or algebraically ghosted, then all the nodal (Lagrange) degrees of freedom associated
    * with the non-local, non-algebraically-ghosted element will be evaluable, and hence that
    * element will be considered evaluable.
+   *
+   * getNonlinearEvaluableElementRange() returns the evaluable element range based on the nonlinear
+   * system dofmap;
+   * getAuxliaryEvaluableElementRange() returns the evaluable element range based on the auxiliary
+   * system dofmap;
+   * getEvaluableElementRange() returns the element range that is evaluable based on both the
+   * nonlinear dofmap and the auxliary dofmap.
    */
   const ConstElemRange & getEvaluableElementRange();
+  const ConstElemRange & getNonlinearEvaluableElementRange();
+  const ConstElemRange & getAuxiliaryEvaluableElementRange();
```

Comment:
So I renamed `getEvaluableElementRange` to `getNonlinearEvaluableElementRange` so that we can be honest about what this method actually does. `getEvaluableElementRange` now uses the multi evaluable predicate to construct an element range that is evaluable based on both the nonlinear dofmap and the aux dofmap.

Just to avoid diffs, I changed all old occurrences of `getEvaluableElementRange` to be `getNonlinearEvaluableElementRange`, although I'm not sure whether that's the original intent.

Reply:
I would prefer to not have `getEvaluableElementRange()` and `getAuxiliaryEvaluableElementRange()` if they are not needed.

Reply:
Yeah, I'll remove `getAuxiliaryEvaluableElementRange()`. Then it's the same issue with `getEvaluableElementRange` -- it is in framework, but I'm testing it in tensor_mechanics. But this time it's hard for me to think of a test case in the framework...

Reply:
I think I would be in favor of lumping all our coverage together for a single `moose` coverage report but I see arguments both ways. It is hard when you have to very creatively invent framework level tests when there is a very natural module use case.